### PR TITLE
Create pipeline from existing code

### PR DESCRIPTION
**Pull request by konrrad**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import argparse
+
+import utils
+from perspective_transform import perspective_transform
+from sheet_cutting import get_corners_of_sheet
+from text_align import align_text
+
+parser = argparse.ArgumentParser()
+parser.add_argument("image_filename")
```
Comment (j-adamczyk) [inline main.py:9]: It's nice to create function like `get_parsed_args()` which creates parser, parses arguments and returns the argparse.Namespace object

File: main.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+def main():
+    args = parser.parse_args()
+    image = utils.read_image_from_file(args.image_filename)
+
+    corners = get_corners_of_sheet(image.copy())
+
+    image = perspective_transform(image.copy(), corners, image.shape)
+    image = align_text(image.copy())
+    utils.show_image(image, "transformed")
```
Comment (j-adamczyk) [inline main.py:20]: Separate main function is not required, it's better to do this in `if __name__ == __main__`

File: sheet_cutting.py [hunk 3]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
        Returns
        -------
        tuple
-       contours
+        contours
        """
     import imutils
```
Comment (j-adamczyk) [inline sheet_cutting.py:34]: This import should not be local, just do global import, or, even better, just copy this function from imutils (it's a thin wrapper library around more standard libraries).

File: sheet_cutting.py [hunk 4]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
...
     """
-       Cuts sheet of paper from the photo
+    Parameters
+    ----------
+    points
+        list of 4 points arranged clockwise
+    Returns
+    -------
+        list
+            points arranged anti-clockwise
```
Comment (j-adamczyk) [inline sheet_cutting.py:47]: Double indent unnecessary, should be:
```
-------
list
    points arranged anti-clockwise
```

File: sheet_cutting.py [hunk 5]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
...
     contours = cv.findContours(canny_output, cv.RETR_TREE, cv.CHAIN_APPROX_SIMPLE)
-    print(type(contours))
     contours = grab_contours(contours)
     min_rects = [cv.minAreaRect(c) for c in contours]
 
     def get_rect_size(rect: np.ndarray):
         # format enforced by opencv
         # RotatedRect:
-        # ((center_x,center_y,(width,heigth),angle)
+        # ((center_x,center_y),(width,height),angle)
```
Comment (j-adamczyk) [inline sheet_cutting.py:80]: Add spaces after commas

File: utils.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    image : numpy.ndarray
+        Original image (np.ndarray).
+
+    Returns
+    -------
+    list
+        List of four points (x, y) arranged anticlockwise,
+        starting top-left corner.
+    """
+    return [(0, 0), (0, image.shape[0] - 1), (image.shape[1] - 1, image.shape[0] - 1), (image.shape[1] - 1, 0)]
```
Comment (j-adamczyk) [inline utils.py:38]: Line >80 characters, please break into 4 lines

File: utils.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def show_image(image: np.ndarray, window_title="image"):
```
Comment (j-adamczyk) [inline utils.py:41]: Multiple arguments should be in separate lines, `window_title` lacks type

File: utils.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    Shows image in a separate window.
+
+    Parameters
+    ----------
+    window_title
```
Comment (j-adamczyk) [inline utils.py:47]: Add `window_type` docstring

File: sheet_cutting.py [hunk 9]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
...
+    points
+        list of 4 points arranged clockwise
+    Returns
+    -------
+        list
+            points arranged anti-clockwise
+    """
+    tmp = points[1].copy()
+    points[1] = points[3]
+    points[3] = tmp
```
Comment (j-adamczyk) [inline sheet_cutting.py:51]: This just swaps `points[1]` and `points[3]`, just use regular Python swap:
```
points[1], points[3] = points[3], points[1]
```

File: sheet_cutting.py [hunk 10]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
...
     return imutils.grab_contours(contours)
 
 
-def cut_sheet_from_image(src: np.ndarray) -> np.ndarray:
+def convert_clockwise_to_anticlockwise(points: list) -> list:
     """
-       Cuts sheet of paper from the photo
+    Parameters
+    ----------
+    points
```
Comment (j-adamczyk) [inline sheet_cutting.py:42]: Lacks type

File: main.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import argparse
+
+import utils
```
Comment (j-adamczyk) [inline main.py:3]: Imports from the project itself should be at the end, so:
```
# Python build-in modules

# 3rd party libraries

# project modules
```

File: main.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def main():
+    args = parser.parse_args()
+    image = utils.read_image_from_file(args.image_filename)
+
+    corners = get_corners_of_sheet(image.copy())
+
+    image = perspective_transform(image.copy(), corners, image.shape)
+    image = align_text(image.copy())
```
Comment (j-adamczyk) [inline main.py:19]: Why image copies?

File: sheet_cutting.py [hunk 13]
```diff
@@ -29,51 +29,57 @@ def grab_contours(contours: tuple) -> tuple:
...
     THRESHOLD = 100  # initial threshold
     canny_output = cv.Canny(src_gray, THRESHOLD, THRESHOLD * 2)
 
+    # contours are arranged clockwise according to documentation
     contours = cv.findContours(canny_output, cv.RETR_TREE, cv.CHAIN_APPROX_SIMPLE)
-    print(type(contours))
     contours = grab_contours(contours)
     min_rects = [cv.minAreaRect(c) for c in contours]
 
     def get_rect_size(rect: np.ndarray):
```
Comment (j-adamczyk) [inline sheet_cutting.py:77]: Lacks docstring

**Review by j-adamczyk — CHANGES_REQUESTED** [review]

File: main.py [hunk 14]
```diff
@@ -1,24 +1,25 @@
+#python built-ins
 import argparse
 
+#project modules
```
Comment (j-adamczyk) [inline main.py]: Comments should start with a space

File: main.py [hunk 15]
```diff
@@ -1,24 +1,25 @@
...
 import utils
 from perspective_transform import perspective_transform
 from sheet_cutting import get_corners_of_sheet
 from text_align import align_text
 
-parser = argparse.ArgumentParser()
-parser.add_argument("image_filename")
 
+def get_parsed_args():
```
Comment (j-adamczyk) [inline main.py:11]: Lacks returned type and docstring

File: main.py [hunk 16]
```diff
@@ -1,24 +1,25 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument("image_filename")
```
Comment (j-adamczyk) [inline main.py:13]: Should have description

File: sheet_cutting.py [hunk 17]
```diff
@@ -18,41 +18,57 @@ def gray_and_blur_image(src: np.ndarray):
     src_gray = cv.blur(src_gray, (3, 3))
     return src_gray
 
+def grab_contours(cnts):
```
Comment (j-adamczyk) [inline sheet_cutting.py:21]: Add `cnts` data type and returned type

**Review by j-adamczyk — APPROVED** [review]

**Review by wojtke — APPROVED** [review]

**Pull request merged**